### PR TITLE
Edit DG

### LINKED ISSUE #137: [PE-D][Tester E] Datetime Expiring
Did a good job checking for interview datetime to be older than the application datetime. A suggestion would be that if the datetime expires (maybe after a given period), the status should change to make it more intuitive for the user.

As shown in the screenshot, the interview datetime is 31-03-2023 14:35 but the current datetime is already 31-03-2023 14:36.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: pzaiming/ped#3

### LINKED ISSUE #140: [PE-D][Tester A] unexpected date inputs
inputting 31-02-2024 for interview date does not give me an error, but the program just silently set it as 29-02-2024. 

Discrepancy with what is input as compared to what is stored might confuse users.

 Program should simply throw an error for an invalidate date like 31-02-2024

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: NappySprout/ped#2

### LINKED ISSUE #151: [PE-D][Tester E] Application Datetime is Missing in the UI
Good job keep track of the application deadline of the applicants which can be useful information!

I am not sure if the team wanted to show the application deadline on the addressbook as this detail can only be seen from using certain commands such as `edit`. If the application datetime is meant to hidden, it should not be revealed at the edit command too as shown in the screenshot. 

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: pzaiming/ped#7

### LINKED ISSUE #154: [PE-D][Tester E] Phone Number as an Attribute for Uniqueness
Good job on preventing the addition of duplicates!

UG mentioned the uniqueness is done through checking name and phone number. However I feel that the phone number should be unique as two different person should not have the same phone number. 

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: pzaiming/ped#8

### PR DESCRIPTION
**Pull request by FredericChow00**
## What
Edit DG according to bugs found and suggested changes to Command section of DG
Added Efforts and Planned Enhancements sections in DG.

## Why
Clarify and proof-read Command section of DG.
Planned Enhancements cover for feature flaws found in PE-D that cannot be changed due to feature freeze.

Fix #137 
Fix #140 
Fix #151 
FIx #154

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1090,9 +1096,49 @@ Expected: All applicants in HMHero are deleted.
...
+In terms of colour scheme, we thought that the dark theme of AB3 was uninspiring and made use of own blue and light colored theme to give this app a nice, warm and welcoming feel. 
+
+
+### Extensive Testing
+
+The increased number of classes, along with all the commands, means that code coverage is likely to decrease and more testing had to be done as compared to that in AB3.
+
+Hence, we took it as an opportunity to do extensive testing for our application through writing effective unit tests for our commands, 
+parsers and model objects. We redesigned how testing is done for commands and manage to cover all commands successfully. 
+We maintained a ~83% code coverage from the original ~72% in AB3, and increased the number of test cases from 248 (in AB3) to 527 (in HMHero).
```
Comment (FredericChow00) [inline docs/DeveloperGuide.md:1125]: will update this number after all testcases are done

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1090,9 +1096,49 @@ Expected: All applicants in HMHero are deleted.
...
+
+## **Planned Enhancements**
```
Comment (FredericChow00) [inline docs/DeveloperGuide.md:1127]: For everyone to add on if they can think of other feature flaws!

We have up to a limit of 10

**Review by FredericChow00 — COMMENTED** [review]
Added some extra comments on my PR

**Review by Merrickneo — APPROVED** [review]
LGTM!

**Pull request merged**